utils/initial_data_preparation.py

```diff
@@ -327,7 +370,7 @@ def get_base_repo_url(url):
     other_domains_df = pd.DataFrame(columns=df.columns)
 
     for domain, domain_df in dfs_by_domain.items():
-        if len(domain_df) > 10:
+        if len(domain_df) >= 10:
             domain_df.to_csv(
                 data_folder / f"{domain.replace('/', '_').replace(':', '_')}.csv",
```

Comment:
How different is this sanitisation of the name from the one performed in: `sanitise_directory_name(...)`? (which you've just added as part of this PR). Would it be useful to have a single utility function that sanitises both values?

Reply:
- Created `utils/string_utils.py`
- amended the scripts `src/github_repo_request_local.py` and `utils/initial_data_preparation.py` accordingly